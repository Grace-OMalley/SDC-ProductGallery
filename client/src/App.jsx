
import React from 'react';
import Gallery from './components/Gallery.jsx';
import axios from 'axios';
class App extends React.Component {
  constructor (props) {
    super(props);
    let productId = new URL(window.location).pathname.split('/').pop() || 1;
    this.state = {
      productId,
      images: []
    };
  }
  componentDidMount() {
    axios.get('/images/' + this.state.productId)
      .then(res => this.setState(res.data))
      .catch(console.error);
  }

  render() {
    return (
      <Gallery images={this.state.images} />
    );
  }
}

export default App;