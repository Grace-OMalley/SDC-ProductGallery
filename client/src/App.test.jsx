import React from 'react';
import renderer from 'react-test-renderer';
import {cleanup, fireEvent, render, queryHelpers, buildQueries, screen} from '@testing-library/react';
import App from './App';
import axios from 'axios';

afterEach(() => {
  cleanup();
  jest.clearAllMocks();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const data = {
  productId: 1,
  images: [1, 2, 3],
  'product_name': 'test'
};

jest.mock('axios');
axios.get.mockImplementation(() => Promise.resolve(data));

it('should make requests for data', () => {
  const {container} = render(<App />);
  expect(axios.get).toHaveBeenCalledTimes(2);
});