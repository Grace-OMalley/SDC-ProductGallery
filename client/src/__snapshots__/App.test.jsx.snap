// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="imageBlock"
>
  <img
    onClick={[Function]}
  />
  <div
    className="imageCaption"
  >
    Click image to open expanded view
  </div>
  <ul
    className="imageRow"
  />
</div>
`;
